cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/RuntimeJobEnvironment.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+import org.apache.twill.api.TwillRunner;
+import org.apache.twill.api.TwillRunnerService;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+import java.util.Map;
+
+/**
+ * Represents runtime job environment that provides information that is needed by the {@link RuntimeJob} to run the job.
+ */
+public interface RuntimeJobEnvironment {
+  /**
+   * Returns a {@link DiscoveryService} for service announcement purpose.
+   */
+  DiscoveryService getDiscoveryService();
+
+  /**
+   * Returns a {@link DiscoveryServiceClient} for service discovery purpose.
+   */
+  DiscoveryServiceClient getDiscoveryServiceClient();
+
+  /**
+   * Returns a {@link TwillRunnerService} for running {@link TwillRunner}.
+   */
+  TwillRunnerService getTwillRunnerService();
```

Comment:
Can it be `TwillRunner` instead? The service expose the start and stop methods, meaning the caller of this method will be responsible for the lifecycle of the service. However, I believe in this case, it is the implementation who manage the lifecycle?

Reply:
yea the lifecycle is managed by extension. So we can make it a runner.